### ISSUE TITLE
Provide options for V2 cli to override the default bundle_url

### ISSUE THREAD

**Issue by Sinofine** [tectonic-typesetting/tectonic#1002]
When we use `tectonic -X new` or `tectonic -X init` to initiate projects, the output is as follows.
```
note: "version 2" Tectonic command-line interface activated
note: creating new document in directory `xxxx`
note: connecting to https://relay.fullyjustified.net/default_bundle.tar
note: resolved to https://data1.fullyjustified.net/tlextras-2020.0r0.tar
```

It seems that the url is hard coded in
https://github.com/tectonic-typesetting/tectonic/blob/8bf8da71424b2c166dbd4e442c908105ed36bce7/crates/bundles/src/lib.rs#L114-L122

and the config is default using that source
https://github.com/tectonic-typesetting/tectonic/blob/8bf8da71424b2c166dbd4e442c908105ed36bce7/src/config.rs#L194-L202

The source listed above is acceptable for daily usage, but for users in an internal network without access to internet, they may use mirrors or local copy for bundles. However, the command `tectonic -X new` or `tectonic -X init` themselves are default using the hard-coded bundle url to complete initiation, which is meant if the default url is not accessible then the initiation won't be accomplished (usually we waits for minutes and then it throws connection errors and fails).

In the V1 cli, it provides `--web-bundle` to override the default, but in the newer V2 cli, the option is absent. If we provide the option for `tectonic -X new` and `tectonic -X init`, it may be nicer for people unable to access the default bundle url.

**Comment by pkgw** [comment]
Yes, this is a good point. This should be a fairly straightforward improvement for someone new to the project to implement, I'd hope, so I'll tag as such!

**Comment by bryango** [comment]
I came up with a temporary fix for this at build time in #1131. Please see if it's okay!
